tensorboard/webapp/widgets/line_chart_v2/lib/renderer/renderer.ts

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {LineSpec, IRenderer} from './renderer_types';
+import {Paths, Rect} from '../types';
+
+export type RenderGroupMap<Cacheable> = Map<
+  string,
+  {
+    data: Paths;
+    cacheable: Cacheable;
+  }
+>;
+
+export abstract class Renderer<Cacheable> implements IRenderer {
```

Comment:
From subclass use, it looks like the mutable render cache map and set of
cache IDs to remove are pretty important. How are subclasses supposed to
use them? Please document.